upgrade_test.py

```diff
@@ -415,186 +415,205 @@ def test_rolling_upgrade(self):  # pylint: disable=too-many-locals,too-many-stat
-                error_factor, schema_load_error_num)
-
-        self.log.debug('start sstabledump verify')
-        self.db_cluster.nodes[0].remoter.run('for i in `sudo find /var/lib/scylla/data/keyspace_complex/ -type f |grep -v manifest.json |grep -v snapshots'
-                                             ' |head -n 1`; do echo $i; sudo sstabledump $i 1>/tmp/sstabledump.output || exit 1; done', verbose=True)
-
-        self.log.info('all nodes were upgraded, and last workaround is verified.')
+        with self.subTest('pre-test - prepare test kesyapces and tables'):
+            # prepare test keyspaces and tables before upgrade to avoid schema change during mixed cluster.
+            self.prepare_keyspaces_and_tables()
+            self.fill_and_verify_db_data('BEFORE UPGRADE', pre_fill=True)
+
+            # write workload during entire test
+            self.log.info('Starting c-s write workload during entire test')
+            write_stress_during_entire_test = self.params.get('write_stress_during_entire_test')
+            entire_write_cs_thread_pool = self.run_stress_thread(stress_cmd=write_stress_during_entire_test)
+
+            # Prepare keyspace and tables for truncate test
+            if self.truncate_entries_flag:
+                self.insert_rows = 10
+                self.fill_db_data_for_truncate_test(insert_rows=self.insert_rows)
+
+        with self.subTest('pre-test - Run stress workload before upgrade'):
+            # complex workload: prepare write
+            self.log.info('Starting c-s complex workload (5M) to prepare data')
+            stress_cmd_complex_prepare = self.params.get('stress_cmd_complex_prepare')
+            complex_cs_thread_pool = self.run_stress_thread(
+                stress_cmd=stress_cmd_complex_prepare, profile='data_dir/complex_schema.yaml')
+
+            # wait for the complex workload to finish
+            self.verify_stress_thread(complex_cs_thread_pool)
+
+            # generate random order to upgrade
+            nodes_num = len(self.db_cluster.nodes)
+            # prepare an array containing the indexes
+            indexes = list(range(nodes_num))
+            # shuffle it so we will upgrade the nodes in a
+            # random order
+            random.shuffle(indexes)
+
+            # prepare write workload
+            self.log.info('Starting c-s prepare write workload (n=10000000)')
+            prepare_write_stress = self.params.get('prepare_write_stress')
+            prepare_write_cs_thread_pool = self.run_stress_thread(stress_cmd=prepare_write_stress)
+            self.log.info('Sleeping for 60s to let cassandra-stress start before the upgrade...')
+            time.sleep(60)
+
+        with DbEventsFilter(type='DATABASE_ERROR', line='Failed to load schema'), \
+             DbEventsFilter(type='SCHEMA_FAILURE', line='Failed to load schema'), \
+             DbEventsFilter(type='DATABASE_ERROR', line='Failed to pull schema'):
```

Comment:
The DbEventsFilter() only works for step 1 - step 3, it might be not enough.
Those error might occur in upgrade of reset nodes (step 5).

Reply:
I saw you added a comment, I will test it the error occur at that time.